docs/qubit-routing/qaoa-qubit-routing.md

```diff
@@ -0,0 +1,125 @@
+- swap_mask (`List[bool]`): This output tags each qubit pair (gate) in the list above as either a `SWAP` gate if the entry is `True` or an `RZZ` gate if the entry is `False`.
+- final_qubit_mapping (`List[int]`): This list reports the final order of the shuffled qubits after the application of `SWAPs`. The list must be labelled with qubit indices from `0` to `n-1`
+
+In order to define such a routing function, one can imagine the following template:
+
+```Python
+def my_custom_routing_function(
+	device: DeviceBase,
+	circuit_to_route: List[List[int]]
+):
+	"""
+	A python implementation of qubit routing for a QAOA circuit cost layer
+	
+	Parameters
+	----------
+	device: DeviceBase
+		OpenQAOA device to run the circuit on
+	circuit_to_route: List[List[int]]
+		The cost layer of QAOA represented as a list of
+		edges to be routed
+
+	Returns
+	-------
+	initial_mapping: List[int]
+		The qubit labels of selected qubits to run the circuit
+	routed_gates: List[List[int]]
+		The routed cost layer of the QAOA circuit
+	swap_mask: List[bool]
+		Label for each gate in the routed_gates list
+		True if its a SWAP, False otherwise
+	final_qubit_mapping: List[int]
+		Qubit order post SWAP application
+	"""
+
+	#my custom routing function
+	return(
+		initial_mapping,
+		routed_gates,
+		swap_mask,
+		final_qubit_mapping
+	)
+```
+
+## Testing your solution with OpenQAOA
+
+To begin testing your routing solution, we can use OpenQAOA workflows. It is quite easy to plug-in your routing solution in the QAOA workflow for a problem of your choice. Presented below, is an example showing the usage of qubit routing within OpenQAOA for an IBMQ quantum computer. 
+
+```Python
+from openqaoa import QAOA
+from openqaoa.problem import MaximumCut
+from openqaoa.backends import create_device
+
+#define a networkx graph for which you want to solve Maxcut
+maxcut = MaximumCut(my_problem_graph).qubo()
+
+#define the Quantum Computer to use for running the QAOA
+
+IBMQ.load_account() #load your IBMQ api_token
+credentials = {
+	hub:'ibmq',
+	group:'open',
+	project:'main'
+}
+device = create_device(location='ibmq', name='ibmq_xyz', **credentials)
+
+#create QAOA workflow and optionally update the properties
+q = QAOA()
+q.set_device(device)
+
+#compile the circuit
+q.compile(problem=maxcut, routing_function = my_custom_routing_function)
+
+#run the optimization loop
+q.optimize()
+```
+
+Additionally, OpenQAOA also lets you visualize the routed circuit by simply calling the visualization methods of the library used for the hardware. Since `qiskit` is used for IBMQ devices, you can visualize your qiskit circuit by calling the `.draw()` method on the compiled circuit.
```

Comment:
Showing how this can be done would be nice. Since getting the circuit is not straight forward.